exporters/prometheus/src/exporter_utils.cc

```diff
@@ -242,40 +247,57 @@ void PrometheusExporterUtils::SetData(std::vector<T> values,
  * Histogram => Prometheus Histogram
  */
 template <typename T>
-void PrometheusExporterUtils::SetData(std::vector<T> values,
-                                      const std::vector<double> &boundaries,
-                                      const std::vector<uint64_t> &counts,
-                                      const metric_sdk::PointAttributes &labels,
-                                      std::chrono::nanoseconds time,
-                                      prometheus_client::MetricFamily *metric_family)
+void PrometheusExporterUtils::SetData(
+    std::vector<T> values,
+    const std::vector<double> &boundaries,
+    const std::vector<uint64_t> &counts,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope,
+    std::chrono::nanoseconds time,
+    prometheus_client::MetricFamily *metric_family)
 {
   metric_family->metric.emplace_back();
   prometheus_client::ClientMetric &metric = metric_family->metric.back();
-  SetMetricBasic(metric, time, labels);
+  SetMetricBasic(metric, time, labels, scope);
   SetValue(values, boundaries, counts, &metric);
 }
 
 /**
  * Set time and labels to metric data
  */
-void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &metric,
-                                             std::chrono::nanoseconds time,
-                                             const metric_sdk::PointAttributes &labels)
+void PrometheusExporterUtils::SetMetricBasic(
+    prometheus_client::ClientMetric &metric,
+    std::chrono::nanoseconds time,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope)
 {
   metric.timestamp_ms = time.count() / 1000000;
 
-  // auto label_pairs = ParseLabel(labels);
+  size_t i = 0;
   if (!labels.empty())
   {
     metric.label.resize(labels.size());
-    size_t i = 0;
     for (auto const &label : labels)
     {
       auto sanitized          = SanitizeNames(label.first);
       metric.label[i].name    = sanitized;
       metric.label[i++].value = AttributeValueToString(label.second);
     }
   }
+  auto scope_name = scope->GetName();
```

Comment:
1. We have to make sure scope is not null before using it. I'd add:

```cc
if (!scope) return;
```

---

The problem you are seeing in the tests is that the `ScopeMetrics::scope_` pointer is uninitialized.... and I am sure it is assumed to be `nullptr` somewhere. Looks like #2301 will fix this. (and it will also give you some merge conflicts): https://github.com/open-telemetry/opentelemetry-cpp/pull/2301/files#diff-e774f313b47bc09b00e03c7d294a3f09fa7c04205b7bb24734f1722030363613

Reply:
I added the suggested line.

Reply:
Even after adding the check for !scope, I am still getting `C++ exception with description "basic_string::_M_create" thrown in the test body.` on this line.  Any suggestions?

Reply:
I was able to check out your PR and run the tests successfully a week ago with these edits. It is too bad that I deleted the branch.

When I try now, it does not work. I do not know if that is from rebasing or changes to this PR.

The `scope_` is still nonsense. It is a raw pointer. Something has to own the object. My guess is that it is being deleted before we try to use it.

Reply:
fixed by https://github.com/open-telemetry/opentelemetry-cpp/pull/2322